### ISSUE TITLE
[bug] running tests fails with a linker error

### ISSUE THREAD

**Issue by twop** [TheLostLambda/knus#11]
Upd: it seems to be there for a while https://users.rust-lang.org/t/linker-dies-when-using-knuffel/101753/3 
Upd2: in my project i was able to **WORK AROUND** the issue by using lld linker like so
```toml
# .cargo/config.toml
[target.aarch64-apple-darwin]
linker = "rust-lld"
```
 

---

 Im on M2 Apple chip, latest xcode command line tools etc, rust version 1.81.0

repro:
```sh
# in the knus repo root
cargo test
```
it seems that the actual test that fails is coming from the readme doc.

```rs
#[derive(knus::Decode)]
enum TopLevelNode {
    Route(Route),
    Plugin(Plugin),
}

#[derive(knus::Decode)]
struct Route {
    #[knus(argument)]
    path: String,
    #[knus(children(name="route"))]
    subroutes: Vec<Route>,
}

#[derive(knus::Decode)]
struct Plugin {
    #[knus(argument)]
    name: String,
    #[knus(property)]
    url: String,
}

fn main() -> miette::Result<()> {
let config = knus::parse::<Vec<TopLevelNode>>("example.kdl", r#"
    route "/api" {
        route "/api/v1"
    }
    plugin "http" url="https://example.org/http"
"#)?;
 Ok(())
 }
```

here is how:
```sh
error: linking with `cc` failed: exit status: 1

--- a bunch of files to link ---

0  0x104d22074  __assert_rtn + 72
          1  0x104c8f994  ld::DynamicAtomFile::makeNamedAtom(std::__1::basic_string_view<char, std::__1::char_traits<char>>, ld::file_format::Scope, bool) + 488
          2  0x104c5f758  ___ZNK2ld10InputFiles11SliceParser15parseObjectFileEPKN6mach_o6HeaderE_block_invoke.48 + 228
          3  0x104c462e4  ld::KindToRelocs::addFixupFromSymbolNum(ld::AtomPlacement const&, ld::DynamicAtom*, unsigned int, unsigned int, long long) const + 116
          4  0x104c5b7f0  ld::InputFiles::SliceParser::parseObjectFile(mach_o::Header const*) const + 8944
          5  0x104c6b830  ld::InputFiles::parseAllFiles(void (ld::AtomFile const*) block_pointer)::$_8::operator()(unsigned long, ld::FileInfo const&) const + 440
          6  0x197000698  _dispatch_client_callout2 + 20
          7  0x197014b9c  _dispatch_apply_invoke3 + 336
          8  0x197000658  _dispatch_client_callout + 20
          9  0x197001ea0  _dispatch_once_callout + 32
          10  0x197013bc8  _dispatch_apply_invoke + 252
          11  0x197000658  _dispatch_client_callout + 20
          12  0x1970124e8  _dispatch_root_queue_drain + 860
          13  0x197012b20  _dispatch_worker_thread2 + 156
          14  0x1971af39c  _pthread_wqthread + 228
          ld: Assertion failed: (name.size() <= maxLength), function makeSymbolStringInPlace, file SymbolString.cpp, line 74.
          clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

my quick search gave two seemingly relevant links:
- https://github.com/leptos-rs/leptos/issues/3148
- https://users.rust-lang.org/t/intel-macos-linking-error/102342

the latter might give a hint why: too much codegen via inlining vs using `Box<dyn Trait>`.

I tried to wrangle the error, but it seems that it is beyond my immediate humble abilities, still wanted to document that issue

**Comment by TheLostLambda** [comment]
Hi @twop ! This is indeed something I've been battling to get working in the CI (it's the reason macos CI is currently disabled)!

It was also an issue with the original `knuffel` repo, and I used the same work-around as you listed above, unfortunately that wasn't seeming to work when I was testing macos builds in CI?

That file is also theoretically already part of this repo: https://github.com/TheLostLambda/knus/blob/main/.cargo/config.toml

In your `Upd2`, were you able to fix the issue on this repo, or was that just you fixing it for another repository that depends on this one?

**Comment by twop** [comment]
that was for the usage in my separate repro project. im wondering though if that setting is honored for doc tests.

**Comment by TheLostLambda** [comment]
Closed by #14
